webapp/src/ts/actions/tasks.ts

```diff
@@ -0,0 +1,81 @@
+import {createAction, Store} from '@ngrx/store';
+import { createSingleValueAction } from './actionUtils';
+
+export const Actions = {
+  setSelectedTask: createSingleValueAction('SET_SELECTED_TASK', 'selected'),
+  setTasksLoaded: createSingleValueAction('SET_TASKS_LOADED', 'loaded'),
+
+  addSelectedTask: createSingleValueAction('ADD_SELECTED_REPORT', 'selected'),
+  removeSelectedTask: createSingleValueAction('REMOVE_SELECTED_TASK', 'id'),
+
+  updateTasksList: createSingleValueAction('UPDATE_TASKS_LIST', 'tasks'),
+  removeTaskFromList: createSingleValueAction('REMOVE_TASK_FROM_LIST', 'task'),
+  resetTasksList: createAction('RESET_TASKS_LIST'),
+};
+
+export class TasksActions {
+  constructor(private store: Store) {}
+
+  setSelectedTask(selected) {
+    return this.store.dispatch(Actions.setSelectedTask(selected));
+  }
+
+  setTasksLoaded(promise) {
+    return this.store.dispatch(Actions.setTasksLoaded(promise));
+  }
+
+
+  addSelectedReport(selected) {
```

Comment:
It looks like you copied the contents of the reports reducer instead of actually migrating the task reducer.